simphony/cuds/abc_dataset.py

```diff
@@ -0,0 +1,174 @@
+        ------
+        ValueError :
+            If any object inside the iterable does not exist.
+        """
+
+    @abstractmethod
+    def get(self, uid):  # pragma: no cover
+        """Returns a copy of the object with the 'uid' id.
+
+        Parameters
+        ----------
+
+        uid : uuid.UUID
+            the uid of the object
+
+        Raises
+        ------
+        KeyError :
+            when the object is not in the container.
+
+        Returns
+        -------
+        object :
+            A copy of the internally stored info.
+        """
+
+    @abstractmethod
+    def remove(self, uids):  # pragma: no cover
+        """Remove the object with the provided uids from the container.
+
+        The uids inside the iterable should exists in the container. Otherwise
+        an exception will be raised.
+
+        Parameters
+        ----------
+        uids : iterable of uuid.UUID
+            the uids of the objects to be removed.
+
+        Raises
+        ------
+        KeyError :
+            If any object doesn't exist.
+        """
+
+    @abstractmethod
+    def iter(self, uids=None, item_type=None):  # pragma: no cover
+        """Generator method for iterating over the objects of the container.
+
+        It can receive any kind of sequence of uids to iterate over
+        those concrete objects. If nothing is passed as parameter, it will
+        iterate over all the objects.
+
+        Parameters
+        ----------
+        uids : iterable of uuid.UUID, optional
+            sequence containing the uids of the objects that will be
+            iterated. When the uids are provided, then the objects are
+            returned in the same order the uids are returned by the iterable.
+            If uids is None, then all objects are returned by the iterable
+            and there is no restriction on the order that they are returned.
+
+        Yields
+        ------
+        object : Particle
+            The object item.
+
+        Raises
+        ------
+        KeyError :
+            if any of the ids passed as parameters are not in the dataset.
+        """
+
+    @abstractmethod
+    def has(self, uid):  # pragma: no cover
+        """Checks if an object with the given uid already exists
+        in the dataset.
+
+        Parameters
+        ----------
+        uid : uuid.UUID
+            the uid of the object
+
+        Returns
+        -------
+        True if the uid is found, False otherwise.
+        """
+
+    @abstractmethod
+    def has_type(self, item_type):  # pragma: no cover
+        """Checks if the specified CUDSItem type is present
+        in the dataset.
+
+        Parameters
+        ----------
+        item_type : CUDSItem
+            The CUDSItem enum of the type of the items to return the count of.
+
+        Returns
+        -------
+        True if the type is present, False otherwise.
+        """
+
+    @abstractmethod
+    def count_of(self, item_type):  # pragma: no cover
```

Comment:
You mean as CUBA keywords? In that case, we'll have to change the implementation and the accepting value. I'd say we have to keep this as is, or we ruin the current interface, so it's just a step in that direction.